### ISSUE TITLE
Question: Developer docs miss deps and ask for old tooling

### ISSUE THREAD

**Issue by cpaelzer** [canonical/ubuntu-pro-client#2892]
Running tox was easy but I stumbled from one rabbit hole to the next.
Here a few suggestions/questions in regard to the developer documentation.

Just running tox got me:

```
...
  building 'apt_pkg' extension
  creating build/temp.linux-x86_64-3.10
  creating build/temp.linux-x86_64-3.10/python
  x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv ...
  In file included from python/acquire-item.cc:24:
  python/apt_pkgmodule.h:14:10: fatal error: apt-pkg/hashes.h: No such file ...
     14 | #include <apt-pkg/hashes.h>
        |          ^~~~~~~~~~~~~~~~~~
  compilation terminated.
  error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
  [end of output]
...
ERROR: could not install deps
```

I can track that down and installing `libapt-pkg-dev` fixed it for me.
Then I realized this comes up over and over again in different contexts :-)
- [missing in 2019](https://github.com/canonical/ubuntu-pro-client/issues/291)
- [added in 2019](https://github.com/blackboxsw/ubuntu-advantage-client/commit/629bc91af093593a33833b7eee23c2bdee556f33)
- [missing again in other context in 2020](https://github.com/canonical/ubuntu-pro-client/issues/1039)
- [or added in 2023 for GH runners](https://github.com/canonical/ubuntu-pro-client/commit/582b5119564151c789b54f7a4131858077c65ec6)
- ... I'll spare you the details

You did the only right thing, having a proper `make deps` based on `d/control` so we can maintain it in one place.
Thanks for that.

But since I wanted to point others at the [developer docs](https://github.com/canonical/ubuntu-pro-client/blob/docs/dev-docs/howtoguides/testing.md) I realized two things we could fix there (or tell me why not).

1. [we still ask them for setup_pyenv.sh](https://github.com/canonical/ubuntu-pro-client/blob/docs/dev-docs/howtoguides/testing.md?plain=1#L7). But that [does not exist since this](https://github.com/canonical/ubuntu-pro-client/commit/b767d63a293a9c8e157b45236f06fe80bcea0a4d#diff-9cea6a1607559a2e7aefc9e317dbd94ee9c9c8488e24f0e86074509cd2850090L15).
2. We do not mention `make deps` before running tox which leads to the error above. Should we mention that?

**Did you check the documentation first?**

After looking for guidance in [the Pro Client documentation](https://canonical-ubuntu-pro-client.readthedocs-hosted.com/en/latest/), mark one of the following checkboxes:

 - [x] There is documentation about this, but it needs improvement
 - [ ] There is no documentation about this

**Comment by s-makin** [comment]
https://github.com/canonical/ubuntu-pro-client/pull/3220

**Comment by lucasmoura** [comment]
This should have been fixed by #3220 
Please let us know if that is not the case and we can reevaluate this issue
